### PR TITLE
Feat: added new CI/CD workflows

### LINKED ISSUE #5: GitHub Actions workflows for desktop/web binaries
Tracking issue for adding workflows to build desktop/web binaries to GitHub Actions.
This should ideally build all branches (except gh-pages) and PRs as well.

Checklist:
- [x] Windows support
- [x] Mac support
- [x] Linux support
- [x] Web support (see [here](https://github.com/enbox/enbox/blob/master/.github/workflows/deploy_web_to_gh_pages.yml#L14) as a guide)

### LINKED ISSUE #8: Keep CNAME on new github pages builds
Currently when the GH pages workflow runs, it overwrites the CNAME file. This is a problem for custom domains which look at the CNAME file to function.

Add the CNAME file to the GitHub pages workflow so that it keeps it on commit.

### PR DESCRIPTION
**Pull request by Treeways**
Closes #5 
Closes #8

Added functionality for building and checking passing/failing builds, as well as web builds on push.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+
+    # Setup the flutter web build
+    - uses: subosito/flutter-action@v1.2.0
+      with:
+        channel: beta
+      if: ${{ runner.os == "Windows" }}
```
Comment (anirudhb) [inline .github/workflows/continuous-integration.yml]: The if and the run should be on the same indent level
like this:
```yaml
- if: runner.os == "Windows"
  run: flutter config --enable-windows-desktop
```
Also note you can drop the `${{ }}` in if statements

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - run: flutter config --enable-windows-desktop
+      if: ${{ runner.os == "macOS" }}
+    - run: flutter config --enable-macos-desktop
+      if: ${{ runner.os == "Linux" }}
+    - run: flutter config --enable-linux-desktop
+    - run: flutter upgrade    
+    if: ${{ runner.os == "Linux" }}
+    - run: flutter config --enable-web
+    - run: flutter upgrade
+    - run: flutter build web --release
```
Comment (anirudhb) [inline .github/workflows/continuous-integration.yml]: You're not making a desktop binary? Also remember to upload the artifact.
And maybe put the `--enable-web` before the upgrade so you don't run it twice.

**Review by anirudhb — DISMISSED** [review]

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+      - name: Configure OS variables
+        - if: ${{ runner.os == Windows }}
```
Comment (anirudhb) [inline .github/workflows/continuous-integration.yml]: Inconsistent indentation. It should be like this instead:
```suggestion
        - name: Configure OS variables
          if: ${{ runner.os == Windows }}
          env:
            OS: windows
```

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Upgrade Flutter and build binaries
+        run: flutter upgrade
+        run: flutter build $OS
+        - if: runner.os == "Linux"
+          run: flutter build web --release
+      - name: Save artifact to display build pass/fail
+        uses: actions/upload-artifact@v1
+        with:
+          name: binary-$OS
+          path: dist
```
Comment (anirudhb) [inline .github/workflows/continuous-integration.yml]: The build directory is called `build`

**Review by anirudhb — CHANGES_REQUESTED** [review]

**Comment by Treeways** [comment]
Closes #8

File: .github/workflows/continuous-integration.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+      - name: Configure OS variables
+        if: ${{ runner.os == Windows }}
+          env:
+            OS: windows
+        if: ${{ runner.os == macOS }}
```
Comment (anirudhb) [inline .github/workflows/continuous-integration.yml]: These have to be different items. It's supposed to be like this:
```suggestion
- if: runner.os == macOS
  env:
    OS: macos

**Review by anirudhb — CHANGES_REQUESTED** [review]

**Review by anirudhb — APPROVED** [review]
Properly formatted

**Pull request merged**